src/main/java/com/flowingcode/vaadin/addons/gridhelpers/LazySelectAllGridHelper.java

```diff
@@ -0,0 +1,50 @@
+package com.flowingcode.vaadin.addons.gridhelpers;
+
+import com.vaadin.flow.component.grid.Grid.SelectionMode;
+import com.vaadin.flow.component.grid.GridMultiSelectionModel;
+import java.io.Serializable;
+
+/**
+ * Vaadin hides select all checkbox when grid's data is NOT in-memory (lazy).
+ * <p>
+ * This grid helper allows showing or hiding select all checkbox no matter if data provider is or
+ * not memory.
+ * <p>
+ * Note that enabling select all checkbox when grid uses a lazy data provider could lead to memory
+ * and performance issues, given that all the items must be eagerly loaded at the moment user clicks
+ * the checkbox.
+ * 
+ * @param <T> the type of items in grid
+ */
+public final class LazySelectAllGridHelper<T> implements Serializable {
```

Comment:
Helper classes must not be public. Any public documentation about the feature goes into `GridHelper`

Reply:
fixed